projects/step-frontend/src/lib/angularjs/js/app.js

```diff
@@ -880,20 +880,28 @@ angular
   })
 
   .service('genericErrorInterceptor', function ($q, $injector) {
-    var service = this;
+    const service = this;
+
+    service.response = function (response) {
+      const Dialogs = $injector.get('Dialogs');
+
+      const responsePayload = response?.data;
+      if (responsePayload?.error) {
+        Dialogs.showErrorMsg(responsePayload.error);
+      }
+
+      return response || $q.when(response);
+    };
+
     service.responseError = function (response) {
-      Dialogs = $injector.get('Dialogs');
-      var responsePayload = response.data;
-      if (response.status != 200 && responsePayload && responsePayload.errorMessage) {
+      const Dialogs = $injector.get('Dialogs');
+      const responsePayload = response?.data;
+      if (response.status !== 200 && responsePayload?.errorMessage) {
         Dialogs.showErrorMsg(responsePayload.errorMessage);
       } else {
         // Legacy error handling
-        if (response.status == 500) {
-          if (
-            responsePayload &&
-            responsePayload.metaMessage &&
-            responsePayload.metaMessage.indexOf('org.rtm.stream.UnknownStreamException') >= 0
-          ) {
+        if (response.status === 500) {
+          if (responsePayload?.metaMessage?.indexOf('org.rtm.stream.UnknownStreamException') >= 0) {
```

Comment:
this could be simplified to ```responsePayload?.metaMessage?.includes('org.rtm.stream.UnknownStreamException')```

Reply:
Since it is legacy code anyway I would just leave it. Hopefully it will be removed soon anyway